openhtmltopdf-core/src/main/java/com/openhtmltopdf/render/BlockBox.java

```diff
@@ -428,6 +430,73 @@ private MarkerData.TextMarker makeTextMarker(LayoutContext c, IdentValue listSty
 
         MarkerData.TextMarker result = new MarkerData.TextMarker();
 
+        result.setAlignment(IdentValue.END);
+        result.setLayoutWidth(w);
+        result.setText(text);
+
+        return result;
+    }
+
+    private MarkerData.TextMarker makeTextMarker(LayoutContext c, CascadedStyle markerStyle) {
+        IdentValue listDirection = getParent().getStyle().getDirection();
+        IdentValue alignment = getStyle().deriveStyle(markerStyle).getTextAlign();
+
+        // coalesce to flow-relative values, for ListItemPainter to use
+        if (listDirection == IdentValue.RTL) {
+            if (alignment == IdentValue.LEFT) {
+                alignment = IdentValue.END;
+            } else {
+                alignment = IdentValue.START;
+            }
+        } else {
+            if (alignment == IdentValue.RIGHT) {
+                alignment = IdentValue.END;
+            } else {
+                alignment = IdentValue.START;
+            }
+        }
+
+        StringBuilder sb = new StringBuilder();
+
+        if (listDirection == IdentValue.RTL) {
+            sb.append("  ");
+        }
```

Comment:
Could you explain what's going on here with the leading and trailing `"  "`?

Reply:
I think it was likely a kludge to ensure the marker had some whitespace padding, but honestly, I can't recall now.

Reply:
thanks